s30020-NikiforKoseda/zad3.txt

```diff
@@ -0,0 +1,131 @@
+Ćwiczenia 1
+Napisz skrypt, który sprawdzi, czy podano więcej niż jeden parametr. Jeśli tak, to niech wyświetli pierwszy parametr. Jeśli nie, to niech wyświetli komunikat, że nie podano parametrów.
+
+###
+#!/bin/bash
+
+if [ "$#" -gt 1 ]; then
+    echo "$1"
+else
+    echo "This script does not have more than one parameter."
+fi
+
+Ten skrypt nauczył mnie ważnej rzeczy - SPACJE SĄ KONIECZNE PRZY NAWIASACH KWADRATOWYCH - inaczej skrypt nie działa. Oraz nauczyłem się -gt, nie można użyć < bo w linuxie to przekierowanie. Oraz $# przechowuje zmienne dla skryptu
+###
+
+Napisz skrypt, który:
+
+sprawdzi, czy istnieje plik config.cfg w bieżącym katalogu.
+
+Jeśli taki plik istnieje, to niech go wykona.
+
+Jeśli nie istnieje, to niech sprawdzi, czy podano argument i wykona plik o nazwie takiej jak wartość argumentu (parametru) skryptu.
+
+Jeśli żadne z powyższych nie nastąpiło, to niech wyświetli komunikat o błędzie.
+
+###
+#!/bin/bashfilename="config.cfg"
+
+if [ -f "$filename" ]; then 
+    source "$filename"
+elif [ "$#" -gt 0 ]; then
+    givenfile="$1"
+    source "$givenfile"
+else
+    echo "Nastąpił bład"
+fi
+
+Przy odpowiedniej konfiguracji można sprawić, aby skrypt podany jako argument przyjmował kolejne argumenty
```

Comment:
W tym zadaniu chodziło o to by zasymulować działanie jakiegoś programu.
Uruchamiamy program. Jeśli nie znajdzie swojego pliku konfiguracyjnego to się uruchamia w sposób standardowy.
Jeśli znajdzie taki plik konfiguracyjny to wtedy ładuje to i uruchamia się z modyfikacjami.
Dla basha jest taki plik konfiguracyjny `~/.bashrc`. Wszystko to co tam się znajdzie będzie automatycznie uruchomione przy każdym uruchomieniu basha (osobne okno, karta, podpowłoka itd...).
Tam można sobie skonfigurować basha na własne potrzeby by np. zmienić sobie PS1 "permanentnie".
Zdefiniować pewne aliasy, które chcemy by było dostępne cały czas itd...

Natomiast samo zadanie wykonałeś poprawnie.